lib/ts/BaseLogFile.h

```diff
@@ -55,16 +53,24 @@ typedef enum {
   LL_Fatal,     // causes process termination
 } LogLogPriorityLevel;
 
-#define log_log_trace(...)                         \
-  do {                                             \
-    if (BASELOGFILE_DEBUG_MODE)                    \
-      BaseLogFile::log_log(LL_Debug, __VA_ARGS__); \
+// log_log_* macros are turned on/off via the BASELOGFILE_DEBUG environment variable.
+// For example, run traffic_server like this: `BASELOGFILE_DEBUG=1 ./traffic_server`
+#define log_log_trace(...)                             \
+  do {                                                 \
+    static bool blf_debug_on = false;                  \
```

Comment:
`log_log_*` functions are used to log interesting bits of information as the actual logging subsystem is coming up.